File: Matlab code/readSSTobservations.m
% File with observation data:
filename = 'obs_sst_ttk17.nc';

% Read time, sst and standard deviations from file (using Matlab's
% simplified NetCDF functions):
time = ncread(filename,'time');
sst = ncread(filename,'sst') - 273.15;
sst_stdev = ncread(filename,'sst_stdev');

% % Read reference time:
% refTimeString = ncreadatt(filename,'time','units');
% refTime = datenum(refTimeString(12:end), 'yyyy-mm-dd');
% time = time + refTime;

cax = [2 9];

% Read SINMOD data:
sFilename = 'E:/nn9828k/spring2021/mids_short/dataset.nc';
ncid = netcdf.open(sFilename);
T = getVariable(ncid, 'temperature', [0 0], [400 350], 0, 0)-273.15;
netcdf.close(ncid);


figure
subplot(2,2,1), pcolor(sst(:,:,1)'), shading flat, colorbar, caxis(cax)
subplot(2,2,2), pcolor(T'), shading flat, colorbar, caxis(cax)
subplot(2,2,3), pcolor(sst_stdev(:,:,1)'), shading flat, colorbar

File: Python code/readSSTobservations.py
import netCDF4 as nc
import matplotlib.pyplot as plt

# File with observation ds:
filename = 'obs_sst_ttk17.nc'

# Read time, sst and standard deviations from file
ds = nc.Dataset(filename)
time = ds.variables['time'][:]
sst = ds.variables['sst'][:] - 273.15
sst_stdev = ds.variables['sst_stdev'][:]

cax = [2, 9]

# Read SINMOD ds:
sFilename = 'dataset.nc'
ds_s = nc.Dataset(sFilename)
T = ds_s.variables['temperature'][0, 0] - 273.15

fig, axs = plt.subplots(2, 2)

# Plot sst
im1 = axs[0, 0].pcolor(sst[0, 0, :, :].T, cmap='jet', vmin=cax[0], vmax=cax[1])
fig.colorbar(im1, ax=axs[0, 0])

# Plot T
im2 = axs[0, 1].pcolor(T.T, cmap='jet', vmin=cax[0], vmax=cax[1])
fig.colorbar(im2, ax=axs[0, 1])

# Plot sst_stdev
im3 = axs[1, 0].pcolor(sst_stdev[0, 0, :, :].T, cmap='jet')
fig.colorbar(im3, ax=axs[1, 0])

plt.show()